### PR TITLE
Add release notes for 1.72.0

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @cuviper @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,100 @@
...
+
+Libraries
+---------
+
+- [Implement `TryFrom<&OsStr>` for `&str`](https://github.com/rust-lang/rust/pull/98202/)
+- [Document memory orderings of `thread::{park, unpark}`](https://github.com/rust-lang/rust/pull/99587/)
+- [io: soften ‘at most one write attempt’ requirement in io::Write::write](https://github.com/rust-lang/rust/pull/107200/)
+- [Specify behavior of HashSet::insert](https://github.com/rust-lang/rust/pull/107619/)
+- [Stabilize `const_cstr_methods`](https://github.com/rust-lang/rust/pull/107624/)
+- [Allow limited access to `OsStr` bytes](https://github.com/rust-lang/rust/pull/109698/)
```
Comment (joshtriplett) [inline RELEASES.md:45]: This isn't stabilized yet, though it's on track to be in a *future* version.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,100 @@
...
+
+Compiler
+--------
+
+- [Remember names of `cfg`-ed out items to mention them in diagnostics](https://github.com/rust-lang/rust/pull/109005/)
+- [Support for native WASM exceptions](https://github.com/rust-lang/rust/pull/111322/)
+- [Add support for NetBSD/aarch64-be (big-endian arm64).](https://github.com/rust-lang/rust/pull/111326/)
+- [Write to stdout if `-` is given as output file](https://github.com/rust-lang/rust/pull/111626/)
+- [Force all native libraries to be statically linked when linking a static binary](https://github.com/rust-lang/rust/pull/111698/)
+- [Support 128-bit atomics on all x86_64 Apple targets](https://github.com/rust-lang/rust/pull/112150/)
```
Comment (taiki-e) [inline RELEASES.md:29]: 128-bit atomics are currently unstable.

**Comment by Mark-Simulacrum** [comment]
Resolved nits & added the stabilized APIs section. Not too many this cycle.

**Pull request merged**